src/imputors/fill.jl

```diff
@@ -34,11 +34,17 @@ end
 # TODO: Switch to using Base.@kwdef on 1.1
 Fill(; value=mean, context=Context()) = Fill(value, context)
 
-function impute!(data::AbstractVector, imp::Fill)
+function _impute!(data::AbstractVector, imp::Fill)
     imp.context() do c
         fill_val = if isa(imp.value, Function)
-            # Call `deepcopy` because we can trust that it's available for all types.
-            imp.value(Impute.drop(data; context=c))
+            available = Impute.drop(data; context=c)
+
+            if isempty(available)
+                @warn "Cannot apply fill function $(imp.value) as all values are missing"
```

Comment:
It is annoying having this at warn level.
Often it is fine. 
So I think it should be at debug level.

One can always use `LoggingExtras.TransformLogger` to bump it up.

Reply:
Sure. I wasn't 100% on the warning anyways.